src/utils/v2/nftRewards.ts

```diff
@@ -0,0 +1,23 @@
+import { NFTRewardTier } from 'models/v2/nftRewardTier'
+
+// Returns the highest NFT reward tier that a payer is eligible given their pay amount
+export function maxEligibleRewardTier({
+  ethPayAmount,
+  nftRewardTiers,
+}: {
+  ethPayAmount: number
+  nftRewardTiers: NFTRewardTier[]
+}) {
+  let nftReward: NFTRewardTier | null = null
+  // all nft's who's thresholds are below the pay amount
+  const eligibleNftRewards = nftRewardTiers.filter(rewardTier => {
+    return rewardTier.paymentThreshold <= ethPayAmount
+  })
+  if (eligibleNftRewards.length) {
+    // take the maximum which is the only one received by payer
+    nftReward = eligibleNftRewards.reduce((prev, curr) => {
+      return prev.paymentThreshold > curr.paymentThreshold ? prev : curr
+    })
+  }
```

Comment:
```suggestion
  const nftRewardTier = nftRewardTiers.filter(rewardTier => {
    return rewardTier.paymentThreshold <= ethPayAmount
  }).reduce((prev, curr) => {
      return prev.paymentThreshold > curr.paymentThreshold ? prev : curr
    })
  }
```

Reply:
<3

Reply:
`.reduce` can't be passed an empty array, need to check the `eligibleNftRewards.length` before calling it

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Reduce_of_empty_array_with_no_initial_value